src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -98,6 +98,21 @@ public static StringComparer Create(CultureInfo culture, bool ignoreCase)
             return new CultureAwareComparer(culture, ignoreCase);
         }
 
+        public static StringComparer Create(CultureInfo culture, CompareOptions options)
+        {
+            if (culture == null)
+            {
+                throw new ArgumentException(nameof(culture));
+            }
+
+            if (!Enum.IsDefined(typeof(CompareOptions), options))
+            {
```

Comment:
we should check the options here as we are doing it in [https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Extensions/src/System/Globalization/Extensions.cs#L29](https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Extensions/src/System/Globalization/Extensions.cs#L29)

the reason is we shouldn't allow ordinal options here

Reply:
why shouldn't we allow ordinal here ?

Reply:
because if anyone want to use ordinal option should create OrdinalComparer and not culture aware comparer 

        private static readonly OrdinalCaseSensitiveComparer s_ordinal = new OrdinalCaseSensitiveComparer();
        private static readonly OrdinalIgnoreCaseComparer s_ordinalIgnoreCase = new OrdinalIgnoreCaseComparer();    

---
In reply to: [167785331](https://github.com/dotnet/coreclr/pull/16334#discussion_r167785331) [](ancestors = 167785331)